### PR TITLE
Add bill tab + update fetchbill

### LINKED ISSUE #78: Implement billing system
And update model.fetchBill() & model.fetchBillList() accordingly.

### LINKED ISSUE #118: [PE-D] Errorneous `bill` command does not provide any message
Aside from the command changing colour to red.

Note: in the picture below, the white text appeared some time before that and persisted. Even though I have typed no fewer than 20 `bill` commands, both valid and invalid.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: wardetu/ped#11

### LINKED ISSUE #132: [PE-D] Bill command not working
Bill command throws errors as well.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: FeliciaTay/ped#12

### LINKED ISSUE #150: [PE-D] bill command not working
The bill command does not display anything despite checking in the guest.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: haveaqiupill/ped#7

### LINKED ISSUE #197: Implement the remaining of fetchbill
- [x] Show total in feedback window
- [x]  Add bill tab to show more details

### PR DESCRIPTION
**Pull request by hmtey**
- [x] Add bill tab
- [x] Store bills
- [x] Refactor code for bill
- [x] Modify fetchbill so that it updates the bill tab view

Details:
1. Added `BookKeeper` class to `Model` for managing transactions
2. Bills can now be saved and read.
3. `CheckIn` command will add a new `Bill` for the room to the `BookKeeper`
4. `CheckOut` command will delete the corresponding `Bill` from `BookKeeper`
4. `FetchBill` command changes the view of the bills tab to show related `Bills`.

Sample bill tab view after `fetchbill i/B000000`
Screenshot 2020-04-11 02 14 37

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(String.format(MESSAGE_NO_BOOKING, personId, roomId));
         }
 
-        model.chargeService(personId, roomId, service.get());
+        model.chargeService(roomId, service.get());
```
Comment (dawidjaja) [inline src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java:84]: Then do you think the person ID parameter is redundant? Is it better to delete it?

File: src/main/java/seedu/address/logic/commands/CheckInCommand.java [hunk 2]
```diff
@@ -84,6 +85,14 @@ public CommandResult execute(Model model) throws CommandException {
         Stay stay = new Stay(person.get(), room.get(), LocalDateTime.now(), toDate, "");
         model.checkIn(stay);
 
+        Optional<Bill> bill = model.findBill(roomId);
+
+        if (bill.isEmpty()) {
+            model.addBill(new Bill(personId, roomId));
+        }
+
+        model.chargeService(roomId, room.get().getRoomCost());
```
Comment (dawidjaja) [inline src/main/java/seedu/address/logic/commands/CheckInCommand.java]: Does this need to be multiplied by the number of nights?

**Review by dawidjaja — CHANGES_REQUESTED** [review]
Hi could you look at these?

File: src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java (hunk 1 above)
Comment (hmtey) [reply src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java:84]: I need to use personId to check if the correct guest is staying in the room via isGuestCheckedIn(Person, Room).

File: src/main/java/seedu/address/logic/commands/CheckInCommand.java (hunk 2 above)
Comment (hmtey) [reply src/main/java/seedu/address/logic/commands/CheckInCommand.java]: Implemented!

**Review by dawidjaja — APPROVED** [review]
LGTM

**Pull request merged**